gdb-server/src/worker.rs

```diff
@@ -124,13 +127,15 @@ pub async fn handler(
     Ok(break_due)
 }
 
-pub async fn await_halt(core: &Core, output_stream: Sender<CheckedPacket>, await_halt: bool) {
-    if await_halt && core.core_halted().unwrap() {
+pub async fn await_halt(core: &Core, output_stream: Sender<CheckedPacket>, await_halt: &mut bool) {
+    task::sleep(Duration::from_millis(10)).await;
+    if *await_halt && core.core_halted().unwrap() {
         let response =
             CheckedPacket::from_data(PacketKind::Packet, "T05hwbreak:;".to_string().into_bytes());
 
         let mut bytes = Vec::new();
         response.encode(&mut bytes).unwrap();
+        *await_halt = false;
```

Comment:
Ah, good catch! I think I did this lateron in the respective handler, which might be too late!

Reply:
Let's see

run correctly sets it to true
stop correctly sets it to false
step correctly sets it to false
user_halt correctly sets it to false

But I don't see it being set after a halt by breakpoint has been performed like here.